packages/devextreme/js/__internal/ui/m_file_uploader.ts

```diff
@@ -861,19 +864,30 @@ class FileUploader extends Editor<FileUploaderProperties> {
     this._isCustomClickEvent = false;
   }
 
-  _attachSelectFileDialogHandler(target) {
+  _selectFileDialogKeyUpHandler(e): void {
+    if (normalizeKeyName(e) === ENTER_KEY) {
+      this._selectFileDialogClickHandler();
+    }
+  }
+
+  _attachSelectFileDialogHandlers(target) {
     if (!isDefined(target)) {
       return;
     }
-    this._detachSelectFileDialogHandler(target);
-    eventsEngine.on($(target), 'click', this._selectFileDialogHandler);
+
+    this._detachSelectFileDialogHandlers(target);
+
+    eventsEngine.on($(target), 'click', this._selectFileDialogClickHandler);
+    eventsEngine.on($(target), 'keyup', this._selectFileDialogKeyUpHandler.bind(this));
   }
 
-  _detachSelectFileDialogHandler(target) {
+  _detachSelectFileDialogHandlers(target) {
     if (!isDefined(target)) {
       return;
     }
-    eventsEngine.off($(target), 'click', this._selectFileDialogHandler);
+
+    eventsEngine.off($(target), 'click', this._selectFileDialogClickHandler);
+    eventsEngine.off($(target), 'keyup', this._selectFileDialogKeyUpHandler.bind(this));
```

Comment:
[mistake]
this is not working while `.bind` creates a new function
Please check how it's done in other components. Probably you can create a new event namespace

Reply:
btw now i see that for avoiding this issue `_selectFileDialogHandler` private cache was used. I suggest you to not use this approach but refactor both old and new subscription

Reply:
I'm assuming the cache was used for a reason. It is needed to ensure that a given handler is removed from the subscription. You suggest using arrow functions, but you can't remove such a subscription because your function is not saved anywhere. You need to pass the same handler (to clear the handlersController) to correctly unsubscribe.

I also see this hack being used in our code: https://github.com/DevExpress/DevExtreme/blob/25_1/packages/devextreme/js/__internal/m_sortable.ts#L66

If I understood correctly, it is not used only where unsubscribe is not important.

This has nothing to do with adding namespace. Nevertheless, I use it to make the code more correct.

Reply:
[6f89b7f67a4112e1f45570a4cfb3a3c4ea8f1247]

Reply:
U can unsubscribe namespace